x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(2);
+      expect(mockLogger.warn).toHaveBeenCalledTimes(0);
+      expect(successfulSingleBulkIndex).toEqual(true);
+    });
+    test('create unsuccessful bulk index due to empty search results', async () => {
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleEmptyDocSearchResults;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+          return false; // value is irrelevant
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(0);
+      expect(mockLogger.error).toHaveBeenCalledTimes(0);
+      expect(successfulSingleBulkIndex).toEqual(true);
+    });
+    test('create unsuccessful bulk index due to bulk index errors', async () => {
+      // need a sample search result, sample signal params, mock service, mock logger
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleDocSearchResultsNoSortId;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const bulkBody = sampleDocSearchResultsNoSortId.hits.hits.flatMap(doc => [
+        {
+          index: {
+            _index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+            _id: doc._id,
+          },
+        },
+        buildBulkBody(doc, sampleParams),
+      ]);
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+
+          // timestamps are annoying...
+          (bulkBody[1] as SignalHit).signal['@timestamp'] = params.body[1].signal['@timestamp'];
+          expect(params.body).toEqual(bulkBody);
+          return {
+            took: 100,
+            errors: true,
+          };
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(2);
+      expect(mockLogger.warn).toHaveBeenCalledTimes(0);
+      expect(mockLogger.error).toHaveBeenCalledTimes(1);
+      expect(successfulSingleBulkIndex).toEqual(false);
```

Comment:
In general for most of our tests we want to have one or maybe at _most_ two or three expects per test. These are pushing the boundary a bit, and you don't have to split these all out, but if you look around the code base most of the examples of unit tests within our plugin, core, and other plugins usually follow the motto of one expect per test section around ~95% of the time.

Exception to the rule would be table test areas where there is a big array and then a `forEach` ontop of that array to test all the values. That is a bit different paradigm but within that we are hopeful that as the table test executes and if it fails it will be one failure per test block still (just wrapped in an array).